packages/neon-api/src/transaction/validator.ts

```diff
@@ -0,0 +1,314 @@
+import { tx, rpc, u, sc } from "@cityofzion/neon-core";
+import { getNetworkFee } from "./util";
+
+export enum ValidationAttributes {
+  None = 0,
+  ValidUntilBlock = 1 << 0,
+  SystemFee = 1 << 1,
+  NetworkFee = 1 << 2,
+  Script = 1 << 3,
+  All = ValidUntilBlock | SystemFee | NetworkFee | Script
+}
+
+export type ValidationSuggestion<T> =
+  | {
+      /**
+       * Whether this is auto-fixed by validator
+       * Validator will try to fix the transaction when param `autoFix` is set to TRUE
+       */
+      fixed: boolean;
+      prev: T;
+      suggestion: T;
+      message?: T;
+    }
+  | {
```

Comment:
Is there any way we can simplify this? Do we have to make `fixed` optional or can we just always return something (`false` if `autoFix` is false?).